### PR TITLE
deposit-form: add interval dates, method and location fields

### PR DESCRIPTION
**Pull request by Glignos**

### CODE REVIEW

File: zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json [hunk 1]
```diff
@@ -1099,5 +1124,172 @@
         ]
       }
     ]
+  },
+  {
+    "type": "fieldset",
+    "title": "Dataset",
+    "description": "Uhm.",
```
Comment (slint) [inline zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json]:
```suggestion
    "description": "Descriptive metadata for the upload, regarding its temporal or geographical attributes.",
```

File: zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json [hunk 2]
```diff
@@ -1099,5 +1124,172 @@
...
+    "fieldhint": "optional",
+    "collapsed": true,
+    "items": [
+      {
+        "type": "array",
+        "key": "locations",
+        "title": "Locations",
+        "fa_cls": "fa-tags",
```
Comment (slint) [inline zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json]: Let's find a decent icon for these guys, e.g. `fa-map-marker` for the locations, maybe a calendar for the dates, etc.

File: zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json [hunk 3]
```diff
@@ -1099,5 +1124,172 @@
...
+                ["Source"], ["Maximize"]
+              ],
+              "extraPlugins": "codesnippet,blockquote,mathjax",
+              "disableNativeSpellChecker": false,
+              "removePlugins": "elementspath",
+              "removeButtons": ""
+            },
+            "fa_cls": "fa-pencil",
+            "minLength": 1,
+            "description": "Collection method."
```
Comment (slint) [inline zenodo/modules/deposit/static/json/zenodo_deposit/deposit_form.json]:
```suggestion
            "description": "Description of the methodology used for the study or research. Optional."
```

**Comment by Glignos** [comment]
Dataset_field_names

**Comment by slint** [comment]
Check the alignment for the elements so that it lines up with the rest of the form inputs:

image

**Pull request merged**